src/tactic/positivity.lean

```diff
@@ -363,4 +361,53 @@ meta def positivity_abs : expr → tactic strictness
   nonnegative <$> mk_app ``abs_nonneg [a] -- else report nonnegativity
 | _ := failed
 
+private lemma nonneg_of_canon [canonically_ordered_add_monoid α] (a : α) : 0 ≤ a := zero_le _
+
+/-- Extension for the `positivity` tactic: Any element of a canonically ordered additive monoid is
+nonnegative. -/
+@[positivity]
+meta def positivity_canon : expr → tactic strictness
+| `(%%a) := nonnegative <$> mk_app ``nonneg_of_canon [a]
+
+private lemma nat_cast_nonneg [ordered_semiring α] (n : ℕ) : 0 ≤ (n : α) := n.cast_nonneg
```

Comment:
This one seems like it doesn't need to exist; it's defeq to `nat.cast_nonneg`, right?